src/submodule.c

```diff
@@ -509,7 +509,7 @@ int git_submodule__map(git_repository *repo, git_strmap *map)
 		data.map = map;
 		data.repo = repo;
 
-		if ((mods = gitmodules_snapshot(repo)) == NULL)
+		if ((error = gitmodules_snapshot(repo, &mods)) < 0 || mods == NULL)
```

Comment:
I feel like this interface has become a bit more brittle by having to check for `mods == NULL`, as well. What about returning for example `ENOTFOUND` in case there are no submodules?

Reply:
I haven't done that in the first place as I then have to reset errors... First lemme see whether this approach will work at all...